tagstudio/src/core/library/alchemy/library.py

```diff
@@ -92,6 +93,33 @@ def get_default_tags() -> tuple[Tag, ...]:
     return archive_tag, favorite_tag
 
 
+@dataclass(frozen=True)
+class SearchResult:
+    """Wrapper for search results.
+
+    :param total_count: total number of items for given query, might be different than len(items)
+    :param items: items for current page (size matches filter.page_size)
+    """
```

Comment:
Just a note, and I discovered one instance of this in #322 as well: Most of the existing codebase is using Google-style docstrings rather than reST, however this is yet to be checked or enforced by a workflow. This is something I'm looking to address in #499, so I'm accepting this as-is and will defer further docstring conversation over to that one.